analyzers/tests/SonarAnalyzer.TestFramework/Verification/DiagnosticVerifier.cs

```diff
@@ -108,7 +108,7 @@ public static class DiagnosticVerifier
             onlyDiagnostics ??= Array.Empty<string>();
             var supportedDiagnostics = analyzer
                 .SelectMany(x => x.SupportedDiagnostics.Select(d => d.Id))
-                .Concat(new[] { AnalyzerFailedDiagnosticId })
```

Comment:
we can Use `.Append(AD001)` here to avoid a new array allocation

Reply:
That is already on the way in #8673, I tried not to modify this method too much here